### ISSUE TITLE
Flaky test: ZooKeeperRegistrationTest.testConnectionRecovery

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#477]
```
ZooKeeperRegistrationTest.testConnectionRecovery: FAILURE
com.linecorp.armeria.server.zookeeper.ZooKeeperRegistrationTest > testConnectionRecovery FAILED
    java.lang.AssertionError: expected:<Expired> but was:<Disconnected>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at com.linecorp.armeria.server.zookeeper.ZooKeeperRegistrationTest.testConnectionRecovery(ZooKeeperRegistrationTest.java:137)
```

**Comment by jonefeewang** [comment]
@anuraaga how to reproduce this?  I have tried many times, can not see this error.

**Comment by trustin** [comment]
Will be fixed by #673
